src/model/Course.java

```diff
@@ -134,79 +153,77 @@ public double getAssignmentNeededGradeAverage() {
 		
 		//return average grade
 		return totalGrade/assignments.size();		
-	}
+	}//end getAssignmentNeededGradeAverage
 	
-	//get 
+	//----------- Get and set methods
 	
-	//Get and set methods
 	//For course ID
-	
 	public int getCourseID() {
 		return courseID;
-	}
+	}//end getCourseID
 	
 	public void setCourseID(int cID) {
 		courseID = cID;
-	}
+	}//end setCourseID
 	
 	//For course code
 	
 	public String getCourseCode() {
 		return courseCode;
-	}
+	}//end getCourseCode
 	
 	public void setCourseCode(String code) {
 		courseCode = code;
-	}
+	}//end setCourseCode
 	
 	//For courseNumber
-	
 	public int getCourseNumber() {
 		return courseNumber;
-	}
+	}//end getCourseNumber
 	
 	public void setCourseNumber(int cNumber) {
 		courseNumber = cNumber;
-	}
+	}//end setCourseNumber
 	
 	//For course name
-	
 	public String getCourseName() {
 		return courseName;
-	}
+	}//end getCourseName
 	
 	public void setCourseName(String cName) {
-		courseName = cName;
-	}
+		//if name is empty, throw error
+		if (cName == null || cName.trim().isEmpty()) {
+            throw new IllegalArgumentException("Course name cannot be empty.");
+        } else {
+    		this.courseName = cName;
+        }//end if-else
+	}//end setCourseName
 	
 	//For course start date
-	
 	public String getCourseStartDate() {
 		return courseStartDate;
-	}
+	}//end getCourseStartDate
 	
 	public void setCourseStartDate(String startDate) {
 		courseStartDate = startDate;
-	}
+	}//end setCourseStartDate
 	
 	//For course end date
-	
 	public String getCourseEndDate() {
 		return courseEndDate;
-	}
+	}//end getCourseEndDate
 	
 	public void setCourseEndDate(String endDate) {
 		courseEndDate = endDate;
-	}
+	}//end setCourseEndDate
 	
 	//For course grade
-	
 	public int getCourseGrade() {
 		return courseGrade;
-	}
+	}//end getCourseGrade
 	
 	public void setCourseGrade(int grade) {
 		courseGrade = grade;
-	}
+	}//end setCourseGrade
 
 }
```

Comment:
Overall, this is great code. I'm so glad to see our application slowly taking shape! You added lots of new functions and features that are very useful, and I see no errors. Great work!